test/runtests.jl

```diff
@@ -54,11 +60,21 @@ const GHA = GitHubActions
 
     @test (@capture_out group(() -> println("!"), "a")) == "::group::a\n!\n::endgroup::\n"
 
-    withenv("a" => nothing) do
-        @test (@capture_out set_env("a", "b")) == "::set-env name=a::b\n"
-        @test ENV["a"] == "b"
-        @test (@capture_out set_env("a", ())) == "::set-env name=a::[]\n"
-        @test ENV["a"] == "[]"
+    mktemp() do file, io
+        withenv("GITHUB_ENV" => file, map(c -> string(c) => nothing, 'a':'z')...) do
+            set_env("a", "b")
+            @test ENV["a"] == "b"
+            @test read(file, String) == "a<<EOF\nb\nEOF\n"
+            set_env("b", "fooEOFbar")
+            @test read(file, String) == "a<<EOF\nb\nEOF\nb<<EOFEOF\nfooEOFbar\nEOFEOF\n"
```

Comment:
I was about to ask what happens in this case, good to see a test for it. But do you know how GitHub would handle it?

Reply:
It should handle any case except for variable names that have `<<` in them, since the delimiter changes to `EOFEOF` if there's an `EOF` in the variable.

For that last `<<` edge case, I actually have no idea but I also know that `@actions/core` doesn't handle it either. I don't even know what "properly" means for that example 😅

Reply:
> I don't even know what "properly" means for that example 😅

I'd guess that's debatable :D 

If this follows `@actions/core`, it's probably fine